src/main/kotlin/com/hypto/iam/server/apis/UsersApi.kt

```diff
@@ -484,3 +486,149 @@ fun Route.createUserPasswordApi() {
+        val pageSize = call.request.queryParameters["page_size"]
+        val sortOrder = call.request.queryParameters["sort_order"]
+        val context =
+            PaginationContext.from(
+                nextToken,
+                pageSize?.toInt(),
+                sortOrder?.let { PaginationOptions.SortOrder.valueOf(it) },
+            )
+        val response = usersService.listMasterUsers(principal, context)
+        call.respondText(
+            text = gson.toJson(response),
+            contentType = ContentType.Application.Json,
+            status = HttpStatusCode.OK,
+        )
+    }
+
+    getWithPermission(
+        listOf(
+            RouteOption(
+                "/organizations/{organization_id}/subordinate_users",
+                resourceNameIndex = 0,
+                resourceInstanceIndex = 1,
+                organizationIdIndex = 1,
+            ),
+        ),
+        "listSubordinateUsers",
+    ) {
+        val organizationId = call.parameters["organization_id"]!!
+        val principal = call.principal<UserPrincipal>()!!
+        require(principal.hrn.organization == organizationId) {
+            "Organization id in path and token are not matching. Invalid token"
+        }
+        val nextToken = call.request.queryParameters["next_token"]
+        val pageSize = call.request.queryParameters["page_size"]
+        val sortOrder = call.request.queryParameters["sort_order"]
+        val context =
+            PaginationContext.from(
+                nextToken,
+                pageSize?.toInt(),
+                sortOrder?.let { PaginationOptions.SortOrder.valueOf(it) },
+            )
+        val response = usersService.listSubordinateUsers(principal, context)
+        call.respondText(
+            text = gson.toJson(response),
+            contentType = ContentType.Application.Json,
+            status = HttpStatusCode.OK,
+        )
+    }
+
+    getWithPermission(
+        listOf(
+            RouteOption(
+                "/organizations/{organization_id}/user_links/{user_link_id}",
+                resourceNameIndex = 2,
+                resourceInstanceIndex = 3,
+                organizationIdIndex = 1,
+            ),
+        ),
+        "switchUser",
+    ) {
+        val organizationId = call.parameters["organization_id"]!!
+        val linkId = call.parameters["user_link_id"]!!
+        val principal = call.principal<UserPrincipal>()!!
+        require(principal.hrn.organization == organizationId) {
+            "Organization id in path and token are not matching. Invalid token"
+        }
+        val response = usersService.switchUser(principal, linkId)
+        call.respondText(
+            text = gson.toJson(response),
+            contentType = ContentType.Application.Json,
+            status = HttpStatusCode.OK,
+        )
+    }
+
+    deleteWithPermission(
+        listOf(
+            RouteOption(
+                "/organizations/{organization_id}/user_links/{user_link_id}",
```

Comment:
Both way deletion must be allowed

Reply:
It is allowed now. But only either leader or the subordinate can delete. No one other than them can delete right now